File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/ColorNode.java
package xyz.wingio.plugins.morehighlight;

import androidx.core.graphics.ColorUtils;

import com.discord.simpleast.core.node.Node;
import com.discord.utilities.textprocessing.*;

import android.text.SpannableStringBuilder;
import android.text.style.*;

public class ColorNode<MessageRenderContext> extends Node<MessageRenderContext> {
  String content;
  int color;

  public ColorNode(String content, int color){
    super();
    this.content = content;
    this.color = color;
  }

  @Override
  public void render(SpannableStringBuilder builder, MessageRenderContext renderContext) {
    int length = builder.length();
    builder.append(content);
    builder.setSpan(new ForegroundColorSpan(color), length, builder.length(), SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE);
    builder.setSpan(new BackgroundColorSpan(ColorUtils.setAlphaComponent(color, 25)), length, builder.length(), SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE);
    builder.setSpan(new StyleSpan(1), length, builder.length(), SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE);
  }
}

File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/node/ColorNode.kt
package xyz.wingio.plugins.morehighlight.node

import android.graphics.Typeface
import android.text.SpannableStringBuilder
import android.text.style.BackgroundColorSpan
import android.text.style.ForegroundColorSpan
import android.text.style.StyleSpan
import androidx.core.graphics.ColorUtils
import com.discord.simpleast.core.node.Node

class ColorNode<MessageRenderContext>(
    private var content: String,
    private var color: Int
): Node<MessageRenderContext>() {

    override fun render(builder: SpannableStringBuilder, renderContext: MessageRenderContext) {
        val length = builder.length
        builder.append(content)

        builder.setSpan(ForegroundColorSpan(color), length, builder.length, SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE)
        builder.setSpan(BackgroundColorSpan(ColorUtils.setAlphaComponent(color, 25)), length, builder.length, SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE)
        builder.setSpan(StyleSpan(Typeface.BOLD), length, builder.length, SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE)
    }

}